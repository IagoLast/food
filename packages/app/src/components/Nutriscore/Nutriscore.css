.Nutriscore {
    padding: 16px;
}

.Nutriscore span {
    color: #888;
    display: block;
    font-weight: 800;
}

.Nutriscore ul {
    display: flex;
    list-style: none;
    padding: 0;
}

.Nutriscore ul li {
    align-items: center;
    color: rgba(255, 255, 255, .6);
    display: flex;
    flex: 1;
    font-size: 2rem;
    font-weight: 600;
    justify-content: center;
    padding: 8px;
    text-align: center;
    vertical-align: center;
    width: 16px;
}

.Nutriscore ul li:first-child {
    border-radius: 8px 0 0 8px;
}

.Nutriscore ul li:last-child {
    border-radius: 0 8px 8px 0;
}

.Nutriscore ul li.a {
    background-color: #048141;
}

.Nutriscore ul li.b {
    background-color: #84bb2f;
}

.Nutriscore ul li.c {
    background-color: #fecb03;
}

.Nutriscore ul li.d {
    background-color: #ee8100;
}

.Nutriscore ul li.e {
    background-color: #e73e10;
}

.Nutriscore ul li.active {
    border-radius: 8px;
    border: 1px solid #555;
    color: white;
    transform: scale(1.2);
}