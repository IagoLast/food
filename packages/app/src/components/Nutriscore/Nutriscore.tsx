import React, { FunctionComponent } from 'react';
import './Nutriscore.css';

interface INutriscoreProps {
    letter: 'a' | 'b' | 'c' | 'd' | 'e';
}

const Nutriscore: FunctionComponent<INutriscoreProps> = ({ letter }) => {
    return (
        <div className="Nutriscore">
            <span>NUTRI-SCORE</span>
            <ul>
                <li className={letter === 'a' ? 'a active' : 'a'}>A</li>
                <li className={letter === 'b' ? 'b active' : 'b'}>B</li>
                <li className={letter === 'c' ? 'c active' : 'c'}>C</li>
                <li className={letter === 'd' ? 'd active' : 'd'}>D</li>
                <li className={letter === 'e' ? 'e active' : 'e'}>E</li>
            </ul>
        </div>
    );
}


export default Nutriscore;