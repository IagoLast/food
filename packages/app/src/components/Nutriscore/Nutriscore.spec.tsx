import Nutriscore from './Nutriscore';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Nutriscore', () => {
    it('should be rendered ok when the "a" score is given', () => {
        const tree = renderer.create(<Nutriscore letter="a" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered ok when the "a" score is given', () => {
        const tree = renderer.create(<Nutriscore letter="b" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered ok when the "a" score is given', () => {
        const tree = renderer.create(<Nutriscore letter="c" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered ok when the "a" score is given', () => {
        const tree = renderer.create(<Nutriscore letter="d" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered ok when the "a" score is given', () => {
        const tree = renderer.create(<Nutriscore letter="e" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});