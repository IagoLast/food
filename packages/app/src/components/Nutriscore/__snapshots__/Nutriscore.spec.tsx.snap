// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nutriscore should be rendered ok when the "a" score is given 1`] = `
<div
  className="Nutriscore"
>
  <span>
    NUTRI-SCORE
  </span>
  <ul>
    <li
      className="active"
    >
      A
    </li>
    <li
      className=""
    >
      B
    </li>
    <li
      className=""
    >
      C
    </li>
    <li
      className=""
    >
      D
    </li>
    <li
      className=""
    >
      E
    </li>
  </ul>
</div>
`;
